packages/graphiql-build-fetcher/src/types.ts

```diff
@@ -0,0 +1,38 @@
+import type { Client } from 'graphql-ws';
+
+/**
+ * Options for creating a sinple, spec-compliant GraphiQL fetcher
+ */
+export interface BuildFetcherOptions {
+  /**
+   * url for HTTP(S) requests. required!
+   */
+  url: string;
+  /**
+   * url for websocket subscription requests
+   */
+  subscriptionsUrl?: string;
```

Comment:
The WebSocket transport supports _all_ GraphQL operations, including streaming queries (like the `@live` or `@deferred` directive).

I guess WS would be used only for subscriptions, which is why the prop makes sense; but, ultimately you can communicate with GQL only through WS if you wish.

Just a thought.

Reply:
yes we're familiar with this pattern, however the intention of this library is to support the more widely adopted transport spec implementations, the "90%+ case" if you will.

Reply:
if someone wants to ship a fetcher for that pattern in a standalone package like this one, they are welcome to

Reply:
Would it be possible to create a simple option for using WebSocket for everything exclusively or an optional function configuration option that receives the operation payload and then depending on the returned value of that function (e.g. `http` or `ws`) the corresponding protocol will be used?

Reply:
folks are welcome to create their own standalone fetcher packages like these for other patterns! we are only interested in adopting widely used, advanced transport specs for this effort